beacon-chain/p2p/custody.go

```
@@ -114,6 +114,61 @@ func (s *Service) UpdateCustodyInfo(earliestAvailableSlot primitives.Slot, custo
 	return earliestAvailableSlot, custodyGroupCount, nil
 }
 
+// UpdateEarliestAvailableSlot updates only the earliest available slot while preserving
+// the current custody group count.
+func (s *Service) UpdateEarliestAvailableSlot(earliestAvailableSlot primitives.Slot) error {
+	// Only update custody info if Fulu is enabled and we're at or past the Fulu fork epoch
+	if !params.FuluEnabled() {
+		return nil
+	}
+
+	currentEpoch := slots.ToEpoch(earliestAvailableSlot)
+	if currentEpoch < params.BeaconConfig().FuluForkEpoch {
+		return nil
+	}
+
+	s.custodyInfoLock.Lock()
+	defer s.custodyInfoLock.Unlock()
+
+	if s.custodyInfo == nil {
```

In this case, should return an error instead.